src/js/core/factories/Grid.js

```diff
@@ -595,10 +595,10 @@ angular.module('ui.grid')
            callback.types.indexOf( type ) !== -1 ||
            type === uiGridConstants.dataChange.ALL ) {
         if (callback._this) {
-           callback.callback.apply(callback._this,this);
+           callback.callback.apply(callback._this, this, options);
```

Comment:
Giving us a way to pass options into a registered callback. This allowed me to prevent buildColumns from being triggered multiple time during initialization. See line 74 of [ui-grid.js](https://github.com/mportuga/ui-grid/blob/75f98c3eda3e46f14f9d924e6f5a6d6e35bcc2b5/src/js/core/directives/ui-grid.js#L74)